solr/core/src/java/org/apache/solr/update/TransactionLog.java

```diff
@@ -239,6 +254,20 @@ public boolean writePrimitive(Object val) throws IOException {
   // for subclasses
   protected TransactionLog() {}
 
+  /**
+   * Sets the counter of written data in the {@link FastOutputStream} view of the log file,
+   * to reflect that we aren't starting at the beginning.
+   */
+  protected void setWrittenCount(long fileStartOffset) throws IOException {
+    fos.setWritten(fileStartOffset);
```

Comment:
If the extending class inserts some metadata in the file header, it needs here to adjust the offsets according to the metadata size.